packages/modules/viewer-react/src/components/app-ui/EcsqlWidget.tsx

```diff
@@ -0,0 +1,141 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+
+import { UiFramework } from "@itwin/appui-react";
+import { QueryRowFormat } from "@itwin/core-common";
+import {
+  Button,
+  ExpandableBlock,
+  LabeledTextarea,
+  Table,
+} from "@itwin/itwinui-react";
+import React, { useState } from "react";
+
+export default function EcsqlWidget() {
+  const [input, setInput] = useState<string>("");
+  const [rows, setRows] = useState<Record<string, string>[]>([]);
+  const [headers, setHeaders] = useState<string[]>([]);
+  const [isloading, setLoading] = useState<boolean>(false);
+  const [error, setError] = useState<string>("");
+
+  const getData = async () => {
+    try {
+      setRows([]);
+      setHeaders([]);
+      setLoading(true);
+      const results = UiFramework.getIModelConnection()!.query(
+        input,
+        undefined,
+        { rowFormat: QueryRowFormat.UseJsPropertyNames }
+      );
+      const rows = [];
+      const headers: string[] = [];
+
+      for await (const obj of results) {
+        const row = { ...obj };
+        // Case 1: Need to convert booleans to strings to display them in the table
+        Object.keys(row)
+          .filter((key) => typeof row[key] === "boolean")
+          .forEach((key) => {
+            row[key] = obj[key].toString();
+          });
+        // Case 2: Need to stringify objects to display them in the table
+        Object.keys(row)
+          .filter((key) => typeof row[key] === "object")
+          .forEach((key) => {
+            row[key] = JSON.stringify(obj[key]);
+          });
+        // Case 3: If the cell content is too long, put it inside an ExpandableBlock
```

Comment:
what does this actually look like?

Reply:
what is the type for row here when this occurs? i dont think it matches what you had originally typed it as

Reply:
Looks like this:
![image](https://user-images.githubusercontent.com/51834103/184981735-8e9fec4b-2d97-4670-96a9-bf02f9e899aa.png)

When expanded:
![image](https://user-images.githubusercontent.com/51834103/184981852-ee7c2288-9f31-437b-a468-b46b6a18ddd3.png)

Reply:
still feels a little weird for me, have you looked at what the standard pattern we use is?
do we just show as much text as we can before showing an ellipses for overflow, and then on hover show the full text in a tool tip or something?

Reply:
iModelConsole shows as much as it can in a cell before showing an ellipses for overflow, and then clicking the cell expands the whole row to display the full text. The Table from core doesn't have anything like that, I think, there's just a button to expand the whole row
![image](https://user-images.githubusercontent.com/51834103/185186799-c0295ac3-31f2-48b3-8906-a79da1231fb7.png)

Reply:
Or maybe I just have to customize it a bit more.